src/solver/conv_hip_implicit_gemm_grouped_wrw_xdlops.cpp

```diff
@@ -400,15 +526,26 @@ bool PerformanceConfigHipImplicitGemmGroupWrwXdlops::SetNextValue(const ProblemD
         assert(!valid_kernels.empty());
         return true;
     }
-    if((index + 1) < valid_kernels.size())
+    do
```

Comment:
not clear as to why you need `do..while(false)` when just adding `{}` suffices.

Reply:
@amberhassaan  It doesnt, notice the `break` statements in there. Its basically a bunch of `goto` but since `use of goto considered harmful` they use the `do ... while(false)` construct

Reply:
I feel more sad :(.

Reply:
Here's an alt version:
```
if (!NextTwoPower...) {
    ...
} else if (!NextLinear..) {
    ...
} else {
   return false;
}
```